### ISSUE TITLE
How do I get the get buildId when I call method buildJob

### ISSUE THREAD

**Issue by mcn259** [bndr/gojenkins#203]

**Comment by WHBANG** [comment]
job, err := Cli.GetJob(JobName)
	if err != nil {
		return 0, err
	}
	b, err := job.GetLastBuild()
	var number int64
	if err != nil {
		number = 1
	} else {
		number = b.GetBuildNumber() + 1
	}
   //(Get the last number) + 1

**Comment by SwenChan** [comment]
Right, but what happen if there is some one has invoked the same job?

**Comment by manilion** [comment]
https://stackoverflow.com/questions/45472604/get-jenkins-job-build-id-from-queue-id

**Comment by StantonMatt** [comment]
Current master has a queue-id path for this, so you do not need to guess the next build number with `GetLastBuild()+1`.

`BuildJob` and `Job.InvokeSimple` return the Jenkins queue item id parsed from the `Location` header. After that, call `GetBuildFromQueueID(ctx, job, queueid)`; it polls the queue item until Jenkins assigns `Executable.Number`, then fetches that build.

The README now shows the same flow:

```go
queueid, err := job.InvokeSimple(ctx, params) // or jenkins.BuildJob(...)
build, err := jenkins.GetBuildFromQueueID(ctx, job, queueid)
```

I rechecked current `master` with a local direct `GetBuildFromQueueID` probe plus the existing focused tests for `BuildJob`, `InvokeSimple`, `Task.Poll`, and `Job.GetBuild`.
